### ISSUE TITLE
vmagent fetching configuration from remote URL shouldn't parse when response code is non-200

### ISSUE THREAD

**Issue by akamensky** [VictoriaMetrics/VictoriaMetrics#4034]
### Describe the bug

When using fetching promscrape configuration from remote URL there is a possibility of internal errors on the other side or invalid URI being requested. Which often results in response code being non-200 (for example 404 or 5xx). However vmagent will still attempt to parse the response contents regardless of content-type and response code. Which results in vmagent crashing with log entry:

```
2023-03-30T04:07:34.200Z	fatal	VictoriaMetrics/lib/promscrape/scraper.go:116	cannot read "http://somefakehost.local/doesntexist.yml": cannot parse Prometheus config from "http://somefakehost.local/doesntexist.yml": cannot unmarshal data: yaml: unmarshal errors:
  line 1: cannot unmarshal !!str `<html> ...` into promscrape.Config
```

This happens even with `-promscrape.config.strictParse=false`

### To Reproduce

1. Have a web server (for example Nginx) serve configuration files
2. Have vmagent request non existent URI on this web server using `-promscrape.config=http://server.local/vmagent-configs/doesntexist.yml`
3. Verify the result

### Version

```
vmagent-20230313-022207-tags-v1.89.1-0-g388d6ee16
```

### Logs

```
2023-03-30T04:07:34.193Z	info	VictoriaMetrics/lib/logger/flag.go:12	build version: vmagent-20230313-022207-tags-v1.89.1-0-g388d6ee16
2023-03-30T04:07:34.193Z	info	VictoriaMetrics/lib/logger/flag.go:13	command-line flags
2023-03-30T04:07:34.194Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -memory.allowedPercent="10"
2023-03-30T04:07:34.194Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -promscrape.config="http://server.local/doesntexist.yml"
2023-03-30T04:07:34.194Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -promscrape.config.strictParse="false"
2023-03-30T04:07:34.194Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -promscrape.configCheckInterval="1m0s"
2023-03-30T04:07:34.194Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.forceVMProto="true"
2023-03-30T04:07:34.194Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.url="secret"
2023-03-30T04:07:34.194Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.vmProtoCompressLevel="8"
2023-03-30T04:07:34.194Z	info	VictoriaMetrics/app/vmagent/main.go:116	starting vmagent at ":8429"...
2023-03-30T04:07:34.194Z	info	VictoriaMetrics/lib/memory/memory.go:42	limiting caches to 13465657753 bytes, leaving 121190919783 bytes to the OS according to -memory.allowedPercent=10
2023-03-30T04:07:34.196Z	info	VictoriaMetrics/lib/persistentqueue/fastqueue.go:59	opened fast persistent queue at "vmagent-remotewrite-data/persistent-queue/1_1F763F8D517D8B17" with maxInmemoryBlocks=11200, it contains 0 pending bytes
2023-03-30T04:07:34.197Z	info	VictoriaMetrics/app/vmagent/remotewrite/client.go:199	initialized client for -remoteWrite.url="1:secret-url"
2023-03-30T04:07:34.197Z	info	VictoriaMetrics/app/vmagent/main.go:142	started vmagent in 0.003 seconds
2023-03-30T04:07:34.197Z	info	VictoriaMetrics/lib/httpserver/httpserver.go:96	starting http server at http://127.0.0.1:8429/
2023-03-30T04:07:34.197Z	info	VictoriaMetrics/lib/httpserver/httpserver.go:97	pprof handlers are exposed at http://127.0.0.1:8429/debug/pprof/
2023-03-30T04:07:34.197Z	info	VictoriaMetrics/lib/promscrape/scraper.go:113	reading Prometheus configs from "http://server.local/doesntexist.yml"
2023-03-30T04:07:34.200Z	fatal	VictoriaMetrics/lib/promscrape/scraper.go:116	cannot read "http://server.local/doesntexist.yml": cannot parse Prometheus config from "http://server.local/doesntexist.yml": cannot unmarshal data: yaml: unmarshal errors:
  line 1: cannot unmarshal !!str `<html> ...` into promscrape.Config
```

### Additional information

Server response to that URI is:

```
$ curl -X GET -i http://server.local/doesntexist.yml
HTTP/1.1 404 Not Found
Server: nginx/1.21.5
Date: Thu, 30 Mar 2023 04:06:43 GMT
Content-Type: text/html
Content-Length: 153
Connection: keep-alive

<html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.21.5</center>
</body>
</html>
```

I believe the vmagent should:

- Verify response code is `200 OK` (at the very least)
- Verify `Content-Type: application/x-yaml` or `Content-Type: text/yaml`, both of which seems to be relatively common to be used for YAML over HTTP

**Comment by valyala** [comment]
vmagent should properly check for HTTP 200 OK response code when fetching config files from http or https urls starting from [v1.87.5 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.87.5). Closing the issue as fixed then.

**Comment by akamensky** [comment]
@valyala Thanks. Will this be released in 1.89.X branch any time soon?

**Comment by valyala** [comment]
> Will this be released in 1.89.X branch any time soon?

The fix will be included in the upcoming `v1.90.0` release. I'll the comment here after the release v1.90.0 will be published. See [the changelog](https://docs.victoriametrics.com/CHANGELOG.html) for details.

**Comment by akamensky** [comment]
Thanks again. No need the comment I'll just track the 1.90 release then.

**Comment by valyala** [comment]
FYI, the bugfix has been also included in [v1.79.12 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.12).

**Comment by valyala** [comment]
FYI, this bugfix has been also included in all the releases starting from [v1.90.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
